tracer/src/Datadog.Trace/Propagators/SpanContextPropagatorFactory.cs

```diff
@@ -13,12 +13,12 @@ namespace Datadog.Trace.Propagators
 {
     internal static class SpanContextPropagatorFactory
     {
-        public static SpanContextPropagator GetSpanContextPropagator(string[] requestedInjectors, string[] requestedExtractors)
+        public static SpanContextPropagator GetSpanContextPropagator(string[] requestedInjectors, string[] requestedExtractors, bool propagationExtractFirst = false)
```

Comment:
Note from our offline chat: Let's remove the default value and make the callers always provide the `propagationExtractFirst` argument. This will require a couple of updates throughout the code but the result will be better in the end